src/spclient.rs

```diff
@@ -56,21 +46,182 @@ pub enum OutputSpendStatus {
     Mined(MinedInBlock),
 }
 
+type WalletFingerprint = [u8; 8];
+
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq)]
 pub struct OwnedOutput {
-    pub txoutpoint: String,
     pub blockheight: u32,
     pub tweak: String,
-    pub amount: u64,
+    pub amount: Amount,
     pub script: String,
     pub label: Option<String>,
     pub spend_status: OutputSpendStatus,
 }
 
+#[derive(Debug, Default, Serialize, Deserialize, Clone, PartialEq)]
+pub struct OutputList {
+    pub wallet_fingerprint: WalletFingerprint,
+    birthday: u32,
+    last_scan: u32,
+    outputs: HashMap<OutPoint, OwnedOutput>,
+}
+
+impl OutputList {
+    pub fn new(scan_pk: PublicKey, spend_pk: PublicKey, birthday: u32) -> Self {
+        // take a fingerprint of the wallet by hashing its keys
+        let mut engine = silentpayments::bitcoin_hashes::sha256::HashEngine::default();
+        engine
+            .write_all(&scan_pk.serialize())
+            .expect("Failed to write scan_pk to engine");
+        engine
+            .write_all(&spend_pk.serialize())
+            .expect("Failed to write spend_pk to engine");
+        let hash = silentpayments::bitcoin_hashes::sha256::Hash::from_engine(engine);
+        let mut wallet_fingerprint = [0u8; 8];
+        wallet_fingerprint.copy_from_slice(&hash.to_byte_array()[..8]);
+        let outputs = HashMap::new();
+        Self {
+            wallet_fingerprint,
+            outputs,
+            birthday,
+            last_scan: if birthday == 0 { 0 } else { birthday - 1 },
```

Comment:
I think it's better to make `last_scan: birthday`. The birthday is set to the current block height when a new user creates a wallet. At that point, the user shouldn't have been able to receive any payments yet, so no need to take that into account here.